test/Sdk.Analyzers.Tests/Sdk.Analyzers.Tests/WebJobsAttributesNotSupportedTests.cs

```diff
@@ -28,21 +28,134 @@ public static void Run([HttpTrigger(AuthorizationLevel.Anonymous, ""get"")] Http
+
+namespace FunctionApp
+{
+    public static class Function1
+    {
+        [Function(""Function1"")]
+        [return: Microsoft.Azure.WebJobs.Queue(""dest-q"")]
+        public static string Run([TimerTrigger(""0 */1 * * * *"")] MyInfo myTimer)
+        {
+            return ""Azure"";
+        }
+    }
+
+    public record MyInfo(bool IsPastDue);
+}";
+            var test = new AnalizerTest
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50.WithPackages(ImmutableArray.Create(
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker", "1.10.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Sdk", "1.7.0"),
+                    new PackageIdentity("Microsoft.Azure.WebJobs.Extensions.Storage", "5.0.1"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Timer", "4.0.1"))),
+
+                TestCode = testCode
+            };
+
+            test.ExpectedDiagnostics.Add(Verify.Diagnostic().WithSeverity(Microsoft.CodeAnalysis.DiagnosticSeverity.Error)
+                .WithSpan(9, 18, 9, 57).WithArguments("QueueAttribute"));
+
+            await test.RunAsync();
+        }
+
+        [Fact]
+        public async Task WebJobsAttributeInReturnTypeAndParameters()
+        {
+            string testCode = @"
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.WebJobs;
+using System.IO;
+
+namespace FunctionApp
+{
+    public static class Function1
+    {
+        [Function(""Function1"")]
+        [return: Microsoft.Azure.WebJobs.Queue(""dest-q"")]
+        public static string Run([TimerTrigger(""0 */1 * * * *"")] object myTimer, 
+                                 [Blob(""samples-workitems/{queueTrigger}"", FileAccess.Read)] Stream myBlob)
+        {
+            return ""Azure"";
+        }
+    }
+}";
+            var test = new AnalizerTest
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50.WithPackages(ImmutableArray.Create(
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker", "1.10.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Sdk", "1.7.0"),
+                    new PackageIdentity("Microsoft.Azure.WebJobs.Extensions.Storage", "5.0.1"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Timer", "4.0.1"))),
+
+                TestCode = testCode
+            };
+
+            // Diagnostic entry for functions parameter
+            test.ExpectedDiagnostics.Add(Verify.Diagnostic().WithSeverity(Microsoft.CodeAnalysis.DiagnosticSeverity.Error)
+                .WithSpan(13, 35, 13, 92).WithArguments("BlobAttribute"));
+            // Diagnostic entry for functions return type
+            test.ExpectedDiagnostics.Add(Verify.Diagnostic().WithSeverity(Microsoft.CodeAnalysis.DiagnosticSeverity.Error)
+                .WithSpan(11, 18, 11, 57).WithArguments("QueueAttribute"));
```

Comment:
Curious, how to find the numbers in WithSpan() method?

Reply:
These 4 values indicates where we are expecting to see the squiggly red line and error.

If I use named arguments for these values,

````
.WithSpan(startLine: 11, startColumn: 18, endLine: 11, endColumn: 57)
````
So this says, the error will be in line 11 of source code and starts at column 18 and ends at 57. This error is on a single line, hence the endLine value is same as startLine.

https://github.com/dotnet/roslyn-sdk/blob/76d342efcf047a014d6d84f87bdd740434b27e1e/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/DiagnosticResult.cs#L310

Reply:
Makes sense. Thanks.